### DISCUSSION TITLE
Downloading Command Line Tools for Xcode

### DISCUSSION THREAD

**Discussion by Moonwalker777** [Homebrew/discussions#131] [Everyday usage] [answered]
Hi,

Im relatively new to GitHub and I need your help.

I tried to install Homebrew by /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)“ this command. But unfortunately I’m stuck at the process of

Downloading Command Line Tools for Xcode

It runs since 15 Minutes and nothing is happening.

Please help :-)

Thank you!

**Comment by gromgit** [comment, marked as answer]
Apple's XCode command-line installer is...quite uncommunicative. You have at least two choices:
1. Wait for as long as it takes...or the installer errors out.
2. Go to https://developer.apple.com/download/more/ (you'll need to sign up for an Apple developer account if you don't have one), then download and install the latest Command Line Tools package [that your OS supports](https://developer.apple.com/support/xcode/#minimum-requirements).

**Reply by Moonwalker777** [reply]
Ok thanks for your help :-)

**Reply by kailynbebee** [reply]
Thank you!

**Reply by lunfy** [reply]
I found that I had to continue the update on Apple's Software Update Tool once that line comes up. It worked for me!

**Reply by allenfinalfinal** [reply]
@lunfy Thanks a million for sharing that! This seems to be working.

**Reply by 0xka13b** [reply]
@lunfy thanks for your answer, checking updates in settings did the trick!

**Reply by Mbutle-ParkingPinLLC** [reply]
> I found that I had to continue the update on Apple's Software Update Tool once that line comes up. It worked for me!

thanks!

**Reply by josh-ryn** [reply]
System Preferences > Software Update --- Update, then Done lol
Was stuck hanging as well. Only waited about 10 minutes until I found this thread. Sped everything up. Thank You!

**Reply by vhnguyenae** [reply]
> I found that I had to continue the update on Apple's Software Update Tool once that line comes up. It worked for me!

lol how did you find it, that did the trick though 🤯

**Reply by D0nFreeman** [reply]
many thanks

**Reply by andrews2k** [reply]
Many thanks !

**Reply by googlecopypasteedit** [reply]
Thanks! Going to the "Software Update" window shows the (painfully slow) download progress.

**Reply by JamesZhao2022** [reply]
Thank you! This really help me for this matter.

**Comment by teddypham9x** [comment]
Just simply wait, at the same time sign up for an Apple developer account :))

**Comment by TylerTemp** [comment]
Actually, I found that, once the download started, you can go to `settings` - `Software Update`, and it will give a very nice view of total size & download speed, with a title saying "XCode Comand-line Tool downloading", and will title switch to "installing" when the download finished.

**Reply by amnbhr** [reply]
nice

**Reply by mafecafe** [reply]
Thank you!!

**Reply by CervellumAI** [reply]
Thanks, never thought about it

**Reply by kennyho94** [reply]
Awesome!

**Reply by shohebislam** [reply]
Thank you so much!

**Reply by MensTekton** [reply]
Oh you mean go to about this mac --> software update.. Thanks!

**Reply by kailynbebee** [reply]
This didn't work for me, FYI! Still got the fix I needed from the chosen answer, but sharing in case anyone else was expecting this thread suggestion to work. I'm on Monterey 12.0.1 if that helps anything.

**Reply by joesebastian6079** [reply]
You are an angel!!

**Reply by LordDashMe** [reply]
Cool!

**Reply by nicolehaug** [reply]
Verified the software update when xcode tool installation hangs is in fact the answer, thx!

**Reply by BrunoVannucci** [reply]
Perfect, after I accessed the Software Update option, I found that it was just waiting for the user to reboot.

**Reply by muescha** [reply]
Consider adding an informative line to the console:

```text
Downloading Command Line Tools for Xcode
  The download duration may vary depending on your bandwidth. 
  You can monitor the progress in Settings under Software Update. 
  To access it, use the command:
open /System/Library/PreferencePanes/SoftwareUpdate.prefPane
```

**Reply by devinvm** [reply]
This is just so wonderful! Great to have such an easy way of accessing a visual progress indicator. Thanks for the tip!

**Reply by zohaibmanzoorkushwahaa** [reply]
this is great it helped me

**Comment by vedant12355** [comment]
you can go to the apple icon in the top left of the menu, then go to about this mac>storage and it should show you the amount of storage space decreasing

**Comment by featuredxp** [comment]
It is like a 450MB package and generally takes longer time than you expect; so give some patience until it is done.
For my side, the patience is ~30 mins.

**Comment by TylerTemp** [comment]
> Oh you mean go to about this mac --> software update.. Thanks!

Also by apple icon (top left of screen) - settings - software update ☺️

**Comment by elias1435** [comment]
it was taken around 10min for me...

**Comment by ghost** [comment]
What a shiet...even Windows updates don't take that long to download...seems like Apple carefully protects their bandwidth -_-

**Comment by gurhub** [comment]
The real scenario is when the install is requested by the terminal, it shows up under the "System Preferences > Software Update" as a new update for the Command Line Tools. I think, the reason might be, that requires an Agreement that you've to accept. Why, because when I pressed the Update it automatically shows up under the terminal as "Downloaded Command Line Tools for Xcode" and "Installing Command Line Tools for Xcode". After that I accepted the agreement then I saw this message "Done with Command Line Tools for Xcode". So, from a software provider company perspective, nothing weird here. But, perhaps there are other delegates methods or messages, whatever it is, that the other side of the development has to follow.

Or, blame it on the boogie ;)

**Comment by ok-coder1** [comment]
I think it is possible to just run `sudo xcode-select --install` and install Homebrew.
